app/models/seller.rb

```diff
@@ -75,26 +75,25 @@ def amount_raised
 
   # calculates the amount raised from gift cards
   def gift_card_amount
-    GiftCardDetail.joins(:item)
-                  .where(items: {
-                           seller_id: id,
-                           refunded: false
-                         })
-                  .inject(0) do |sum, gift_card|
-      sum + gift_card.amount
-    end
+    GiftCardDetail
+      .joins(:item)
+      .where(items: {
+               seller_id: id,
+               refunded: false
+             })
+      .joins("join (#{GiftCardAmount.latest_amounts_sql}) as la on la.gift_card_detail_id = gift_card_details.id")
+      .sum(:value)
   end
 
   # calculates the amount raised from donations
   def donation_amount
-    DonationDetail.joins(:item)
-                  .where(items: {
-                           seller_id: id,
-                           refunded: false
-                         })
-                  .inject(0) do |sum, donation|
-      sum + donation.amount
-    end
+    DonationDetail
+      .joins(:item)
+      .where(items: {
+               seller_id: id,
+               refunded: false
+             })
+      .sum(:amount)
```

Comment:
Does this sum in SQL as opposed to in rails?

Reply:
Yup! Makes it into `SELECT SUM(amount)`--which I believe should be more efficient here as the query is now retreiving only one value, instead of the set which we then have to sum in code